### PR TITLE
Resolves some more build warnings on library project

### PR DESCRIPTION
**Pull request by valadas**
I would use some advice on correct replacement in EventManager.cs for the deprecated EventLogController.EventLogType
and
EventLogController.Instance.AddLog call if anyone has some idea on how to refactor that properly. If not, then let's not block this PR for just this and revisit this part later.

### CODE REVIEW

**Review by bdukes — APPROVED** [review]
I've adjusted the EventManager usages, let me know if that looks right to you

File: DNN Platform/Library/Common/Initialize.cs [hunk 1]
```diff
@@ -33,34 +36,33 @@ public class Initialize
...
-        /// -----------------------------------------------------------------------------
         /// <summary>
         /// Inits the app.
         /// </summary>
         /// <param name="app">The app.</param>
-        /// -----------------------------------------------------------------------------
         public static void Init(HttpApplication app)
         {
             string redirect;
+            var status = Globals.DependencyProvider.GetRequiredService<IApplicationStatusInfo>().Status;
```
[the remark this answers is not in the document]
Comment (bdukes) [reply DNN Platform/Library/Common/Initialize.cs:48]: I think this should be fine, just want to make sure…

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Library/Common/Initialize.cs:48]: Ok, thanks, I will give it a test and report back...

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Library/Common/Initialize.cs:48]: I have installed Dnn from a build on this branch, nothing unexpected logged in Admin logs or server logs, clear cache and restart application work too. So looks fine to me...

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Library/Common/Initialize.cs:48]: Looks ok to me but it makes me wonder about IEventLogger. So if someone Inherits IEventLogger and implements it, then they would get the obsolete usage implemented and they change it. But at the same time, changing the interface would be a breaking change. I am wondering how we migrate that properly...

I love what you did with the parameterless constructor :) 

But yeah it makes we wonder how we gonna revamp Interfaces in a fashion that allows us to have deprecations in a version and removal in 2 versions from now... Should we start making numbered interfaces like IEventLogger2 ?

Thank beeing said, with the current set of changes, I am good with that for now.

[the remark this answers is not in the document]
Comment (bdukes) [reply DNN Platform/Library/Common/Initialize.cs:48]: `IEventLogger` doesn't depend on any obsolete types.  The problem of deprecating and changing interfaces is a big one that we need to think through, but I don't think it applies to `IEventLogger` at this point.

**Review by david-poindexter — APPROVED** [review]

**Pull request merged**